python/cudf/cudf/core/series.py

```diff
@@ -812,6 +797,20 @@ def __getitem__(self, arg):
         else:
             return self.loc[arg]
 
+    def __iter__(self):
+        cudf.utils.utils.raise_iteration_error(obj=self)
+
+    iteritems = __iter__
```

Comment:
If we don't intend to support it, then we shouldn't have it. `__iter__` is an exception since we don't want to allow the fallback to `__getitem__`.

Reply:
Ditto for `items` below